### DISCUSSION TITLE
Adding hash pow to messages as non-spam signal

### DISCUSSION THREAD

**Discussion by FreeTrade** [farcasterxyz/protocol#40] [FIP Stage 1: Ideas]
Seems like there is a danger that the system will be overwhelmed by spam/flood messages if they are free to produce. This may give rise to centralization problems as seen with e-mail where a small number of hubs (like gmail, outlook) decide for the whole network what is allowed and which hubs are legit.
Another solution to spam/flooding might be to require each message to have a minimum amount of hashing done on the message as a signal the message is costly ( see hashcash by Adam Back) - I would suggest a CPU friendly hashing algo, RandomX maybe.

**Comment by varunsrin** [comment]
Yes, I agree with that concern - the first step we've taken is limiting the amount of data each user can store by implementing [pruned sets](https://hackmd.io/fCa8_RCEQ4qBYZjfnas9Zg). This reduces the total storage requirement but does not solve the general network traffic problem of having a lot of messages. 

@cassonmars has suggested using VDFs in the past, I like that approach since it aligns more closely with what we want to ensure. we might consider such a design for the 2.1 release.
